File: API/Section/Summary/C#/Summary/StartForm.cs
using System;
using System.Windows.Forms;
using System.Xml;

namespace GrapeCity.ActiveReports.Samples.CalculatedFields
{
	/// <summary>
	/// CalculatedFields sample shows how to create a new Calculated Field in an ActiveReport and use it with the built in summary features.
	/// </summary>
	public partial class StartForm : System.Windows.Forms.Form
	{
		public StartForm()
		{
			InitializeComponent();
			comboBox1.Items.AddRange(new[] { Properties.Resources.OrdersReport, Properties.Resources.DataFieldExpressionsReport });
		}

		/// <summary>
		/// The main entry point for the application.
		/// </summary>
		[STAThread]
		static void Main() 
		{
			Application.Run(new StartForm());
		}

		
		private void Button_Click(object sender, EventArgs e)
		{
			var rpt = new SectionReport();
			var reportPath = "..\\..\\" + (string)comboBox1.SelectedItem;
			rpt.LoadLayout(XmlReader.Create(reportPath));
			rpt.PrintWidth = 6.5F;
			arvMain.LoadDocument(rpt);
		}

		private void comboBox1_SelectedIndexChanged(object sender, EventArgs e)
		{
			if (!button1.Enabled && comboBox1.SelectedItem != null)
			{
				button1.Enabled = true;
			}
		}

		private void comboBox1_KeyPress(object sender, KeyPressEventArgs e)
		{
			e.Handled = true;
		}
	}
}

File: API/Section/Summary/VB.NET/Summary/StartForm.vb
Imports System.Xml

'CalculatedFields sample shows how to create a new Calculated Field in an ActiveReport and use it with the built in summary features.
Public Class StartForm
	Inherits System.Windows.Forms.Form

	Dim rpt As New SectionReport
	Public Sub New()
		MyBase.New()

		'This call is required by the Windows Form Designer.
		InitializeComponent()
		Me.comboBox1.Items.AddRange(New String() {My.Resources.OrdersReport, My.Resources.DataFieldExpressionsReport})

	End Sub

	'Form overrides dispose to clean up the component list.
	Protected Overloads Overrides Sub Dispose(ByVal disposing As Boolean)
		If disposing Then
			If Not (components Is Nothing) Then
				components.Dispose()
			End If
		End If
		MyBase.Dispose(disposing)
	End Sub

	'The main entry point for the application.
	<STAThread()>
	Shared Sub Main()
		Application.Run(New StartForm())
	End Sub 'Main



	Private Sub Button_Click(sender As Object, e As EventArgs) Handles button1.Click
		Dim rpt As New SectionReport
		Dim reportPath As String = "..\\..\\" + CType(comboBox1.SelectedItem, String)
		rpt.LoadLayout(XmlReader.Create(reportPath))
		rpt.PrintWidth = 6.5!
		arvMain.LoadDocument(rpt)
	End Sub

	Private Sub comboBox1_SelectedIndexChanged(sender As Object, e As EventArgs) Handles comboBox1.SelectedIndexChanged
		If (Not button1.Enabled AndAlso Not IsNothing(comboBox1.SelectedItem)) Then
			button1.Enabled = True
		End If
	End Sub

	Private Sub comboBox1_KeyPress(sender As Object, e As KeyPressEventArgs) Handles comboBox1.KeyPress
		e.Handled = True
	End Sub
End Class
